### PR TITLE
Implement ixfr fallback and unify file and auto for transfering

### PR DESCRIPTION
**Pull request by miekg**
Signed-off-by: Miek Gieben <miek@miek.nl>

### CODE REVIEW

**Comment by miekg** [comment]
Fixes: #2585

File: plugin/auto/README.md [hunk 1]
```diff
@@ -76,3 +75,8 @@ org {
     }
 }
 ~~~
+
+## Also
```
Comment (SuperQ) [inline plugin/auto/README.md]: Minor nit, I'm not sure "Also" is a good section header.

File: plugin/etcd/xfr.go [hunk 2]
```diff
@@ -1,25 +1,17 @@
...
 )
 
-// Serial implements the Transferer interface.
+// Serial returns the serial number to use.
 func (e *Etcd) Serial(state request.Request) uint32 {
 	return uint32(time.Now().Unix())
 }
 
-// MinTTL implements the Transferer interface.
+// MinTTL returns the minimal TTL.
```
Comment (SuperQ) [inline plugin/etcd/xfr.go:14]: I know this is just comment docs cleanup, but is there any documentation on why this is the way it is?

File: plugin/etcd/xfr.go [hunk 3]
```diff
@@ -1,25 +1,17 @@
...
-	"context"
 	"time"
 
 	"github.com/coredns/coredns/request"
-
-	"github.com/miekg/dns"
 )
 
-// Serial implements the Transferer interface.
+// Serial returns the serial number to use.
```
Comment (SuperQ) [inline plugin/etcd/xfr.go]:
```suggestion
// Serial returns the current time as the zone serial number to use.
```

File: plugin/file/setup.go [hunk 4]
```diff
@@ -21,26 +21,43 @@ func setup(c *caddy.Controller) error {
 		return plugin.Error("file", err)
 	}
 
-	// Add startup functions to notify the master(s).
+	f := File{Zones: zones}
+	// get the transfer plugin, so we can send notifies and send notifies on startup as well.
```
Comment (SuperQ) [inline plugin/file/setup.go]:
```suggestion
	// Get the transfer plugin, so we can send notifies and send notifies on startup as well.
```

**Review by SuperQ — COMMENTED** [review]
Some minor docs nit picks.

File: plugin/auto/README.md (hunk 1 above)
Comment (miekg) [reply plugin/auto/README.md]: fixed

File: plugin/etcd/xfr.go (hunk 2 above)
Comment (miekg) [reply plugin/etcd/xfr.go:14]: you mean why '30'? I can't recall, maybe SkyDNS did this

**Comment by miekg** [comment]
ok, a bit hellish to rebase this - conflicts everywhere and too many commits. I'm going to export to the diff and apply it to master and make a new PR

**Comment by miekg** [comment]
has been merged in a new pr

**Pull request closed without merging**